### DISCUSSION TITLE
How to get WorkflowInstanceId in HTTPResponse

### DISCUSSION THREAD

**Discussion by enycaltran** [elsa-workflows/elsa-core#5021] [Q&A]
Hello, I'm trying to get WorkflowInstanceId and ActivityInstanceId in order to expose them in HTTP Response content.
So, designing workflow with Elsa Studio, I wrote:

Variables:
InputDaata:  {{ Variables.InputData.Name }} / {{ Variables.InputData.Email }} 
"WWFII": {{ WorkflowInstanceId }}
"AII": {{ ActivityInstanceId }}

in HTTP Response content (interpreted with Liquid).

But what I actually obtain, is this string:

Variables:
InputDaata:  John Doe / john.doe@acme.com 
"WWFII": 
"AII":

So, what is the right way to access to WorkflowInstanceId  value?

Thank you,
Enrico.

**Comment by glime-ai** [comment]
To access the `WorkflowInstanceId` in an HTTP Response content using Liquid syntax within Elsa Studio, you should be able to use `{{ WorkflowInstanceId }}` directly without the need to reference it through the `Variables` object. The `WorkflowInstanceId` is a property of the workflow context and should be accessible directly in your Liquid template.

However, the `ActivityInstanceId` is not directly accessible as a global property like `WorkflowInstanceId`. To include the `ActivityInstanceId` in your HTTP Response, you would need to set it as a variable within the activity's execution context in your workflow definition.

Here's how you can set it up:

1. In your activity where you want to capture the `ActivityInstanceId`, use a script or a custom activity to set the `ActivityInstanceId` to a variable.
2. Then, in your HTTP Response activity, you can reference this variable using Liquid syntax.

For example, if you set a variable named `CurrentActivityInstanceId` in your workflow, you would access it in your HTTP Response content like this:

```liquid
"WWFII": {{ WorkflowInstanceId }}
"AII": {{ CurrentActivityInstanceId }}
```

Make sure that the variable `CurrentActivityInstanceId` is set before the HTTP Response activity is executed.

If you are still facing issues, please ensure that you are using the correct version of Elsa and that the syntax you are using is compatible with that version. Also, check the documentation for any version-specific changes to the way variables and workflow context properties are accessed.

**Reply by enycaltran** [reply]
If you look well, in order to get WorkflowInstanceId I did exactly what you suggested, writing 

{{ WorkflowInstanceId }}

but I obtained string empty.

So the suggested solution doesn't work.
